### DISCUSSION TITLE
Enable Yubico in Vaultwarden

### DISCUSSION THREAD

**Discussion by Rubensky1** [dani-garcia/vaultwarden#3248] [General]
Hello,

I try to enable 2FA with my Yubikey in Vaultwarden. I recieve this error massage:

I run Vaultwarden on my Synology nas in Docker.
I try to enable the Yubico functions to make a taks in the task scheduler. The script i use is this:

docker run -d --name Vaultwarden
-e YUBICO_CLIENT_ID=****
-e YUBICO_SECRET_KEY=mysecrey
-v /volume1/Docker_Data/Vaultwarden:/data
-p 80:80
vaultwarden/server:latest

It don’t work because the task scheduler give the error the name Vaultwarden is already in use for a container.

We can help me to fix this?
![Vaultwarden](https://user-images.githubusercontent.com/125509993/219163464-1812f767-ec61-44bd-8602-285b778c6c33.png)

I also have start an topic in this place. https://vaultwarden.discourse.group/t/enable-yubico-in-vaultwarden/2380

**Comment by BlackDex** [comment]
I suggest to take a look at: https://github.com/dani-garcia/vaultwarden/wiki/Updating-the-vaultwarden-image

Just as a reference:
- https://github.com/dani-garcia/vaultwarden/wiki/Starting-a-Container
- https://github.com/dani-garcia/vaultwarden/wiki/Enabling-Yubikey-OTP-authentication

**Reply by Rubensky1** [reply]
Thanks for your answer. My container and Vaultwarden works fine. I only try to enable 2FA with my yubikey. 
The script i use (post #1) works fine. when i use a container name that not exist i can activate my Yubikey. The problem is after that the container is broken.

**Reply by BlackDex** [reply]
I understand that, that is why i suggested to read the Updating wiki link.
The reason it doesn't work for you is because you need to remove the container. Only stopping it doesn't work, since that, only stops the container. You can't start two containers with the exact same name. That link I provided shows on how to do that.

Updating the container version or changing `-e` options need the same procedure.

**Comment by Rubensky1** [comment]
Yesterday late i understand that i have to remove the container before running the script. That part is working for now. But when i configure the Yubikey. It take a very lont time before i can hit save. After i click save notting happen. Sometimes i see the error OTP provided. Most of the time nothing,

In the Docker logging i can't find anything that helps me out. 
Maybe you can help?

**Reply by BlackDex** [reply]
Please go to the `/admin/diagnostics` and check if you have a working internet connection within the pod/container.
That is probably the issue.

**Reply by BlackDex** [reply]
Not sure if you have some strange characters there? You could try to use something simple.
The JSON file is also not needed and should **not** be used to configure Vaultwarden. The `-e` should be enough.
So. `-e ADMIN_TOKEN="MijnGeheimeToken123"` should work just fine.

**Comment by Rubensky1** [comment]
1 now i have the admin page online. No internet access. When i disable my firewall i have internet access. But that not a solution. Inside of the firewall i make i rule access al for Docker vault. Unfortunately it didn't work. How must i configure the firewall to have internet access?

**Comment by Rubensky1** [comment]
1Fix the firewall problem. Give my iprange access in the firewall.

**Comment by Rubensky1** [comment]
New question about the admin page. 

For security reasons i want to disable the admin page. 

I remove the admin_token line in config.json. no result.
option 2: i Run the script:
docker run -d --name Vaultwarden \
  -e DISABLE_ADMIN_TOKEN=true \
  -v /volume1/Docker_Data/Vaultwarden:/data/ \
  -p 80:80 \
  vaultwarden/server:latest

No result.

Option 3 i make a line in the config.json
"disable_admin_token": true,

No result.

I have restart my container after changes.
How can i disable the admin page?

**Reply by stefan0xC** [reply]
@Rubensky1 Be aware that [disabling the admin token](https://github.com/dani-garcia/vaultwarden/wiki/Disable-admin-token) by setting `DISABLE_ADMIN_TOKEN=true` will make the admin interface available to anyone!

As explained [in the wiki](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page) you'll have to "_make sure no `ADMIN_TOKEN` environment variable is set, and no `"admin_token"` key exists in `config.json`, if that file exists_" to disable the admin panel.

Furthermore like @BlackDex [already explained](https://github.com/dani-garcia/vaultwarden/discussions/3248#discussioncomment-4988369) just restarting the container will not have any effect as environment variables are bound to the container on creation.

**Comment by Rubensky1** [comment]
Thanks. I remove the container once more. Now the admin page is disabled.  :)
